File: 1. Learn the Basics/3. Recursion/Java/3. Print Linearly From N to 1.java
/* package codechef; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */

/*
Print linearly from n to 1 using Recursion
*/

class Codechef
{
    public static void printNum(int n){
        if(n==0){
            return;
        }
        System.out.println(n);
        printNum(n-1);
    }
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		printNum(n);
	}
}

File: 3. Recursion/C++/3. Print Linearly From N to 1.cpp
/*
Print linearly from n to 1 using Recursion
*/

#include <bits/stdc++.h>
using namespace std;

void printNum(int n){
    if(n==0){
        return;
    }
    cout << n << endl;
    printNum(n-1);
}

int main() {
	// your code goes here
    int n;
    cin >> n;
    printNum(n);
	return 0;
}
